### ISSUE TITLE
Pass decoder to callback function instead of opaque 'context'?

### ISSUE THREAD

**Issue by wziard** [FLIF-hub/FLIF#493]
The callback function for progressive display gets passed an opaque 'context' which it can use to generate a preview. However, if you are loading an animation you need to decoder (or FLIF_info) as well to get the number of images.

The viewflif program 'solves' this by creating a global variable holding the decoder. I currently solve it by adding the decoder to my user_data.  Wouldn't it be better to just pass a FLIF_DECODER to the callback function and make flif_decoder_generate_preview() take a FLIF_DECODER * instead of a void *?

**Comment by jonsneyers** [comment]
Makes sense. I don't currently have time to make these changes, but if anyone feels like making a pull request... go ahead!
